pytest/lib/mocknet.py

```diff
@@ -327,13 +388,244 @@ def stop_node(node):
         m.run('sudo -u ubuntu -i', input=TMUX_STOP_SCRIPT)
 
 
+def upload_and_extract(node, compressed_filename, dst_filename, src_filename):
+    node.machine.upload(compressed_filename,
+                        '/home/ubuntu',
+                        switch_user='ubuntu')
+    node.machine.run(
+        'bash',
+        input=
+        f'unzip /home/ubuntu/$(basename {compressed_filename}); mv /home/ubuntu/$(basename {src_filename}) {dst_filename}'
```

Comment:
Doesn’t `node.machine.run` already run command through shell? So this should just be: `node.machine.run('unzip ...; mv ...')`, no?  Also, why call `basename` on remote side? Just use `os.path.basename`? And like before perhaps use `shlex`. Lastly, it feels like it would be easier to use `gzip` since you wouldn’t need to do `mv` afterwards; just upload directly as `{dst_filename}.gz`.

Reply:
Thanks for the `gzip` suggestion.